### PR TITLE
boards: arm: Add support for Arduino Nicla Sense ME

### PR DESCRIPTION
**Pull request by benjaminbjornsson**
This commit adds support for the Arduino Nicla Sense ME board.
The board functionality has been tested with various samples
such as hello_world, philosophers and peripheral_dis.

Signed-off-by: Benjamin Björnsson <benjamin.bjornsson@gmail.com>

### CODE REVIEW

File: boards/arm/arduino_nicla_sense_me/Kconfig [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Arduino Nicla Sense ME board configuration
```
Comment (tejlmand) [inline boards/arm/arduino_nicla_sense_me/Kconfig:1]: This file is optional, so as it is anyway empty it can simply be removed.

**Review by tejlmand — CHANGES_REQUESTED** [review]
Thanks for this contribution.

A minor request for removing an optional empty file.

File: boards/arm/arduino_nicla_sense_me/Kconfig (hunk 1 above)
Comment (benjaminbjornsson) [reply boards/arm/arduino_nicla_sense_me/Kconfig]: Yes, thank you 👍

File: boards/arm/arduino_nicla_sense_me/board.cmake [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# SPDX-License-Identifier: Apache-2.0
+
+set(OPENOCD_NRF5_SUBFAMILY "nrf52")
```
Comment (mbolivar-nordic) [inline boards/arm/arduino_nicla_sense_me/board.cmake:3]: Why is this here if you are not including openocd support? Please either remove this line or add openocd support to the board.

File: boards/arm/arduino_nicla_sense_me/arduino_nicla_sense_me_defconfig [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+CONFIG_HW_STACK_PROTECTION=y
+
+# enable GPIO
+CONFIG_GPIO=y
+
+# enable uart driver
+CONFIG_SERIAL=y
+
+# enable sam-ba bootloader
+CONFIG_BOOTLOADER_BOSSA=y
```
Comment (mbolivar-nordic) [inline boards/arm/arduino_nicla_sense_me/arduino_nicla_sense_me_defconfig:18]: board.cmake is only declaring support for pyocd, so why is this enabling bossa here? I would have expected support for the bossac runner enabled in board.cmake.

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]
Thanks for your contribution! I noticed a few issues, but I'd be glad to see support for this board added.

File: boards/arm/arduino_nicla_sense_me/board.cmake (hunk 2 above)
Comment (benjaminbjornsson) [reply boards/arm/arduino_nicla_sense_me/board.cmake:3]: I probably thought it had some effect on the pyocd... I'll just remove that line since it works without it 👍

File: boards/arm/arduino_nicla_sense_me/arduino_nicla_sense_me_defconfig (hunk 3 above)
Comment (benjaminbjornsson) [reply boards/arm/arduino_nicla_sense_me/arduino_nicla_sense_me_defconfig:18]: To my understanding bossa doesn't support SAMD11 so I'll just remove this line 👍

File: boards/arm/arduino_nicla_sense_me/board.cmake (hunk 2 above)
Comment (benjaminbjornsson) [reply boards/arm/arduino_nicla_sense_me/board.cmake:3]: Regarding support for openocd, I just can't get anything other than pyocd working.

**Review by mbolivar-nordic — DISMISSED** [review]

**Comment by benjaminbjornsson** [comment]
@tejlmand ping

**Review by tejlmand — DISMISSED** [review]
lgtm

**Comment by benjaminbjornsson** [comment]
Is there anything I should do about the failed test in order for this to get merged? Seems to me like the it's unrelated to my pull request?

**Comment by MaureenHelm** [comment]
> Is there anything I should do about the failed test in order for this to get merged? Seems to me like it's unrelated to my pull request?

Please rebase to pick up the fix

**Comment by benjaminbjornsson** [comment]
Should I re-request review?

**Review by tejlmand — DISMISSED** [review]
lgtm - re-approved

**Review by gmarull — CHANGES_REQUESTED** [review]
All in-tree boards have now been ported to pinctrl, so please do the same here. You can refer to https://github.com/zephyrproject-rtos/zephyr/pull/43926 for more details.

**Comment by carlescufi** [comment]
> All in-tree boards have now been ported to pinctrl, so please do the same here. You can refer to #43926 for more details.

Sorry for this @benjaminbjornsson, your PR landed in the middle of a pinctrl transition.

**Comment by gmarull** [comment]
You can try with:

```
python scripts/utils/pinctrl_nrf_migrate.py -i boards/arm/arduino_nicla_sense_me/arduino_nicla_sense_me.dts --no-backup --header $'/*\n * Copyright (c) 2022 Benjamin Björnsson <benjamin.bjornsson@gmail.com>.\n * SPDX-License-Identifier: Apache-2.0\n */\n\n'
```

and add `CONFIG_PINCTRL=y` to the board defconfig

**Review by carlescufi — APPROVED** [review]
LGTM now, thank you!

**Review by gmarull — APPROVED** [review]

**Comment by carlescufi** [comment]
@benjaminbjornsson ignore the compliance check failure, it's unrelated to your patches.

**Comment by gmarull** [comment]
looks like you'll need to rebase after https://github.com/zephyrproject-rtos/zephyr/pull/44042 (hotfix)...
